include/deal.II/base/function.templates.h

```diff
@@ -783,6 +783,93 @@ VectorFunctionFromTensorFunction<dim, RangeNumberType>::vector_value_list(
 }
 
 
+
+template <int dim, typename RangeNumberType>
+FunctionFromFunctionObjects<dim, RangeNumberType>::FunctionFromFunctionObjects(
+  const unsigned int n_components,
+  const double       initial_time)
+  : Function<dim, RangeNumberType>(n_components, initial_time)
+  , function_values(n_components)
+  , function_gradients(n_components)
+{}
+
+
+
+template <int dim, typename RangeNumberType>
+FunctionFromFunctionObjects<dim, RangeNumberType>::FunctionFromFunctionObjects(
+  const std::vector<std::function<RangeNumberType(const Point<dim> &)>> &values,
+  const double initial_time)
+  : Function<dim, RangeNumberType>(values.size(), initial_time)
+  , function_values(values)
+  , function_gradients(values.size())
+{}
+
+
+
+template <int dim, typename RangeNumberType>
+FunctionFromFunctionObjects<dim, RangeNumberType>::FunctionFromFunctionObjects(
+  const std::vector<std::function<RangeNumberType(const Point<dim> &)>> &values,
+  const std::vector<
+    std::function<Tensor<1, dim, RangeNumberType>(const Point<dim> &)>>
+    &          gradients,
+  const double initial_time)
+  : Function<dim, RangeNumberType>(values.size(), initial_time)
+  , function_values(values)
+  , function_gradients(gradients)
+{}
+
+
+
+template <int dim, typename RangeNumberType>
+RangeNumberType
+FunctionFromFunctionObjects<dim, RangeNumberType>::value(
+  const Point<dim> & p,
+  const unsigned int component) const
+{
+  AssertIndexRange(component, this->n_components);
+  Assert(function_values[component],
+         ExcMessage("Empty lambda for component values"));
```

Comment:
would an error message like "Accessing value() in FunctionFromFunctionObjects requires setting the std::function objects for the value" or something like that be better?